src/AggregateFunctions/AggregateFunctionSumMap.h

```diff
@@ -250,35 +264,86 @@ class AggregateFunctionMapBase : public IAggregateFunctionDataHelper<
         }
     }
 
-    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf) const override
+    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf, std::optional<size_t> version) const override
     {
+        if (!version)
+            version = getDefaultVersion();
+
         const auto & merged_maps = this->data(place).merged_maps;
         size_t size = merged_maps.size();
         writeVarUInt(size, buf);
 
+        std::function<void(size_t, const Array &)> serialize;
+        switch (*version)
+        {
+            case 0:
```

Comment:
This could be a fix for https://github.com/ClickHouse/ClickHouse/issues/16587#issuecomment-726713143, but not sure for now - it will require to add some more if conditions to make this fix fully correct, but this will slow down the serialization. So it will be moved to some new "testing" aggregate function just for tests if nothing good comes to mind.